### PR TITLE
Migrate from Travis to GH actions

### PR DESCRIPTION
**Pull request by nicolo-ribaudo**
| Q                        | A 
| ------------------------ | ---
| License                  | MIT

I got an email from travis a few days ago that they "migrated us to the new plan", and now Travis isn't running on our PRs anymore.
When I check why, I get this error:
> Builds have been temporarily disabled for public repositories due to a negative credit balance.

Context : https://blog.travis-ci.com/2020-11-02-travis-ci-new-billing

We _could_ write them an email asking for some free credits and their blog post says that they would probably give us those credits, but:
- We would have to do this periodically, every time we run out of credits
- A fellow maintainer with a ton of popular OSS repositories tried contacting them and their support was unresponsive

<a href="https://gitpod.io/#https://github.com/babel/babel/pull/12412"></a>

### CODE REVIEW

**Review by JLHwung — APPROVED** [review]
LGTM if CI is green.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -139,3 +163,64 @@ jobs:
...
+          make build-plugin-transform-runtime-dist
+      - name: Test on Windows
+        # Hack: --color has supports-color@5 returned true for GitHub CI
+        # Remove once `chalk` is bumped to 4.0.
+        run: |
+          ./node_modules/.bin/jest --ci --color
+        env:
+          BABEL_ENV: test
+
+  external-parser-tests:
```
Comment (JLHwung) [inline .github/workflows/ci.yml:199]: Not a blocker: The external parser tests is pretty fast (~1m), I would consider we merge these three tests into a single runner for less resources overhead.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -93,6 +93,30 @@ jobs:
...
+        uses: actions/checkout@v2
+      - name: Use Node.js latest
+        uses: actions/setup-node@v2-beta
+      - name: Get yarn cache directory path
+        id: yarn-cache-dir-path
+        run: echo "::set-output name=dir::$(yarn config get cacheFolder)"
+      - uses: actions/cache@v2
+        with:
+          path: ${{ steps.yarn-cache-dir-path.outputs.dir }}
+          key: ${{ runner.os }}-yarn-${{ hashFiles('yarn.lock') }}
```
Comment (merceyz) [inline .github/workflows/ci.yml:111]: The yarn cache is actually identical on all OSes so you can have all the jobsshare the same cache

Comment (nicolo-ribaudo) [reply .github/workflows/ci.yml:111]: Oh thanks! I left it with the os name because I wasn't sure about it, but I'll update now.

**Comment by nicolo-ribaudo** [comment]
I also cleaned up `Makefile` by removing now unused commands (because they are split across different jobs and/or steps).

**Review by JLHwung — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -139,3 +163,65 @@ jobs:
...
+        with:
+          name: babel-artifact
+      - name: Generate runtime helpers
+        run: |
+          make build-plugin-transform-runtime-dist
+      - name: Test on Windows
+        # Hack: --color has supports-color@5 returned true for GitHub CI
+        # Remove once `chalk` is bumped to 4.0.
+        run: |
+          ./node_modules/.bin/jest --ci --color
```
Comment (merceyz) [inline .github/workflows/ci.yml:195]:
```suggestion
        run: yarn jest --ci --color
```
Right?

Comment (JLHwung) [reply .github/workflows/ci.yml:195]: It is intentional. We need to run Jest on node.js 6.

Comment (nicolo-ribaudo) [reply .github/workflows/ci.yml:195]: I copied this from an existing job introduced inhttps://github.com/babel/babel/commit/c1f205038e9a5bea9971335334bd63b2d57291c6. @JLHwung do you remember if there is a reason for using `./node_modules/.bin/jest`?

Comment (JLHwung) [reply .github/workflows/ci.yml:195]: We could use `yarn` here since we are not running Windows test on Node.js 6.

**Review by existentialism — APPROVED** [review]

**Comment by merceyz** [comment]
@nicolo-ribaudo Any plans on moving the circle ci part to GH actions as well? Would probably save a minute each time the lockfile changes :slightly_smiling_face:

**Pull request merged**